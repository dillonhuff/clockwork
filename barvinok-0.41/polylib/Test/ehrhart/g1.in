# Gauss Anticipated memory
#
# lambda = ( 1 1 1 )
#
# sigma = ( 1 0 0 )
#         ( 0 1 0 )

# P = a1, Q = a2
# R = i0, S = j0, T = k0
# U = N
#  i  j  k  P  Q  R  S  T  U  cte
7 11
0  1  0 -1  0  0  0  0  0  0  -1   # i = k+1
0  0  1 -1  0  0  0  0  0  0  -1   # j = k+1
1  0  0  1  0  0  0  0  0  0  -1   # k >= 1
1  0  0 -1  1  0  0  0  0  0   0   # k <= a1
1  0  0 -1  0  1  0  0  0  0   0   # k <= a2
1  0  0  1  1  1 -1 -1 -1  0  -1   # a1 + a2 + k >= i0 + j0 + k0 + 1
1  0  0 -1 -1 -1  0  0  1  2   1   # a1 + a2 + k <= k0 + 2 N + 1

# 6 params
#  P  Q  R  S  T  U cte
11 8
1  0  0  0  0  0  1  0   # N >= 0
1  0  0  1  0 -1  0  0   # i0 >= k0
1  0  0 -1  0  0  1  1   # i0 <= N + 1
1  0  0  0  1 -1  0  0   # j0 >= k
1  0  0  0 -1  0  1  0   # j0 <= N
1  0  0  0  0  1  0 -1   # k0 >= 1
1  0  0  0  0 -1  1  0   # k0 <= N
1  1  0  0  0 -1  0  0   # a1 >= k0
1 -1  0  0  0  0  1  1   # a1 <= N + 1
1  0  1  0  0 -1  0  0   # a2 >= k
1  0 -1  0  0  0  1  0   # a2 <= N

