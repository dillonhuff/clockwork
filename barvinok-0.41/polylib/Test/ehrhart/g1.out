---------------------------------------
Domain:
         2P + Q - T -2U  -1 >= 0
         P + 2Q - T -2U  -1 >= 0
         P + Q - R - S - T  >= 0
         - P + U + 1 >= 0
         T  -1 >= 0
         - Q + U  >= 0
         R - T  >= 0
         - R + U + 1 >= 0
         S - T  >= 0
         - S + U  >= 0
          1 >= 0

Vertices:
[ -P-Q+T+2U+2,  -P-Q+T+2U+2,  -P-Q+T+2U+1 ]
[ 2,  2,  1 ]

Ehrhart Polynomial:
( -1 * P + ( -1 * Q + ( 1 * T + ( 2 * U + 1 )
 )
 )
 )

---------------------------------------
Domain:
         2P + Q - T -2U  -1 >= 0
         P + 2Q - T -2U  -1 >= 0
         - P - Q + R + S + T  >= 0
         - R - S + 2U  >= 0
         - P + U + 1 >= 0
         T  -1 >= 0
         - Q + U  >= 0
         R - T  >= 0
         - R + U + 1 >= 0
         S - T  >= 0
         - S + U  >= 0
          1 >= 0

Vertices:
[ -P-Q+T+2U+2,  -P-Q+T+2U+2,  -P-Q+T+2U+1 ]
[ -P-Q+R+S+T+2,  -P-Q+R+S+T+2,  -P-Q+R+S+T+1 ]

Ehrhart Polynomial:
( -1 * R + ( -1 * S + ( 2 * U + 1 )
 )
 )

---------------------------------------
Domain:
         - P - Q + R + S + T  >= 0
         -2P - Q + T + 2U + 1 >= 0
         - P + Q  >= 0
         P - T  >= 0
         T  -1 >= 0
         - Q + U  >= 0
         R - T  >= 0
         - R + U + 1 >= 0
         S - T  >= 0
         - S + U  >= 0
         2P + Q - R - S - T  -1 >= 0
          1 >= 0

Vertices:
[ -P-Q+R+S+T+2,  -P-Q+R+S+T+2,  -P-Q+R+S+T+1 ]
[ P+1,  P+1,  P ]

Ehrhart Polynomial:
( 2 * P + ( 1 * Q + ( -1 * R + ( -1 * S + ( -1 * T + 0 )
 )
 )
 )
 )

---------------------------------------
Domain:
         - P - Q + R + S + T  >= 0
         - P -2Q + T + 2U + 1 >= 0
         P - Q  >= 0
         P + 2Q - R - S - T  -1 >= 0
         Q - T  >= 0
         - P + U + 1 >= 0
         T  -1 >= 0
         - Q + U  >= 0
         R - T  >= 0
         - R + U + 1 >= 0
         S - T  >= 0
         - S + U  >= 0
          1 >= 0

Vertices:
[ -P-Q+R+S+T+2,  -P-Q+R+S+T+2,  -P-Q+R+S+T+1 ]
[ Q+1,  Q+1,  Q ]

Ehrhart Polynomial:
( 1 * P + ( 2 * Q + ( -1 * R + ( -1 * S + ( -1 * T + 0 )
 )
 )
 )
 )

---------------------------------------
Domain:
         - P -2Q + T + 2U + 1 >= 0
         P - Q  >= 0
         P + Q - R - S - T  >= 0
         Q - T  >= 0
         - P + U + 1 >= 0
         T  -1 >= 0
         - Q + U  >= 0
         R - T  >= 0
         - R + U + 1 >= 0
         S - T  >= 0
         - S + U  >= 0
          1 >= 0

Vertices:
[ Q+1,  Q+1,  Q ]
[ 2,  2,  1 ]

Ehrhart Polynomial:
( 1 * Q + 0 )

---------------------------------------
Domain:
         -2P - Q + T + 2U + 1 >= 0
         - P + Q  >= 0
         P + Q - R - S - T  >= 0
         P - T  >= 0
         T  -1 >= 0
         - Q + U  >= 0
         R - T  >= 0
         - R + U + 1 >= 0
         S - T  >= 0
         - S + U  >= 0
          1 >= 0

Vertices:
[ P+1,  P+1,  P ]
[ 2,  2,  1 ]

Ehrhart Polynomial:
( 1 * P + 0 )

